### ISSUE TITLE
issues with tama_remove_fragment_models.py 

### ISSUE THREAD

**Issue by ariadnamorales** [GenomeRIK/tama#35]
Hello,
I am trying several ways to filter my transcripts, and I have the following issues:

 - issue 1
I am using tama_remove_fragment_models.py with the output file of tama_cds_regions_bed_add.py, but the bed file is not returning the full label "G1;G1.20;Q3ZCT1;full_length;90_match;prot_ok;F2" it is only returning something like "G1;G1.20". Having the full label is useful when looking at the files in Genome Browser. Is the script only keeping "full_length;full_match"?

- issue 2
With the same data mentioned above, I ran tama_remove_polya_models_levels.py, then tama_remove_fragment_models.py, and then the ORF_NMD scripts. However, while trying to run the last step with tama_cds_regions_bed_add.py. I get the error 
"Mismatching lengths!
G1;G1.83	10763	8511"

Please, advise!

Thanks,

Ariadna

**Comment by GenomeRIK** [comment]
Hi Ariadna,

Glad to see you are using a lot of the more specialized TAMA tools!

Regarding your issues:

issue 1
I am using tama_remove_fragment_models.py with the output file of tama_cds_regions_bed_add.py, but the bed file is not returning the full label "G1;G1.20;Q3ZCT1;full_length;90_match;prot_ok;F2" it is only returning something like "G1;G1.20". Having the full label is useful when looking at the files in Genome Browser. Is the script only keeping "full_length;full_match"?

You may be using an older version of TAMA but I have added a parameter to let you use the original ID structure form the input BED file:

  -id ID      Use original ID line original_id (Default is tama_id line based on gene_id;transcript_id structure

So you might want to re-clone and try this feature.

issue 2
With the same data mentioned above, I ran tama_remove_polya_models_levels.py, then tama_remove_fragment_models.py, and then the ORF_NMD scripts. However, while trying to run the last step with tama_cds_regions_bed_add.py. I get the error
"Mismatching lengths!
G1;G1.83 10763 8511"

I suspect that the BED file you are using for "tama_cds_regions_bed_add.py" is not the same file that was produced after running "tama_remove_polya_models_levels.py". Can you check that the input BED file is the same?

If you use the BED file from before running "tama_remove_polya_models_levels.py", then the issue is that the ID's are changed to keep the consecutive gene count numbers. So the ID's are no longer matching between the ones used for the "tama_orf_seeker.py" tool and the ones used for adding the CDS regions into the BED file. 

So basically, can you check that the "tama_cds_regions_bed_add.py" input BED file is the same one you used to generate the fasta file that was the input for running "tama_orf_seeker.py"?

If this is not the issue, then there is some other issue with the ID's that we will need to look into further. 

Cheers,
Richard

**Comment by GenomeRIK** [comment]
Hi Ariadna,

I am going to close this issue now but if you are still having problems feel free to open it again. 

Cheers,
Richard

**Comment by GenomeRIK** [comment]
Hi Ariadna,

I recently got an email from another user who had some issues with the output from tama_remove_fragment_models.py not working with the ORF/NMD pipeline. I think maybe the issue you had may be related to this. I have now updated the repo to fix these issues so hopefully the problem is resolved. 

Thank you,
Richard
